test/parallel/test-runner-force-exit.mjs

```diff
@@ -0,0 +1,84 @@
+import * as common from '../common/index.mjs';
+import * as fixtures from '../common/fixtures.mjs';
+import * as snapshot from '../common/assertSnapshot.js';
+import { describe, it } from 'node:test';
+import { hostname } from 'node:os';
+import { fileURLToPath } from 'node:url';
+
+function replaceTestDuration(str) {
+  return str
+    .replaceAll(/duration_ms: [0-9.]+/g, 'duration_ms: *')
+    .replaceAll(/duration_ms [0-9.]+/g, 'duration_ms *');
+}
+
+const root = fileURLToPath(new URL('../..', import.meta.url)).slice(0, -1);
+
+const color = '(\\[\\d+m)';
+const stackTraceBasePath = new RegExp(`${color}\\(${root.replaceAll(/[\\^$*+?.()|[\]{}]/g, '\\$&')}/?${color}(.*)${color}\\)`, 'g');
+
+function replaceSpecDuration(str) {
+  return str
+    .replaceAll(/[0-9.]+ms/g, '*ms')
+    .replaceAll(/duration_ms [0-9.]+/g, 'duration_ms *')
+    .replace(stackTraceBasePath, '$3');
+}
+
+function replaceJunitDuration(str) {
+  return str
+    .replaceAll(/time="[0-9.]+"/g, 'time="*"')
+    .replaceAll(/duration_ms [0-9.]+/g, 'duration_ms *')
+    .replaceAll(hostname(), 'HOSTNAME')
+    .replace(stackTraceBasePath, '$3');
+}
+
+function removeWindowsPathEscaping(str) {
+  return common.isWindows ? str.replaceAll(/\\\\/g, '\\') : str;
+}
+
+function replaceTestLocationLine(str) {
+  return str.replaceAll(/(js:)(\d+)(:\d+)/g, '$1(LINE)$3');
+}
+
+const defaultTransform = snapshot.transform(
+  snapshot.replaceWindowsLineEndings,
+  snapshot.replaceStackTrace,
+  removeWindowsPathEscaping,
+  snapshot.replaceFullPaths,
+  snapshot.replaceWindowsPaths,
+  replaceTestDuration,
+  replaceTestLocationLine,
+);
+
+const transformers = {
+  junit: snapshot.transform(
+    replaceJunitDuration,
+    snapshot.replaceWindowsLineEndings,
+    snapshot.replaceStackTrace,
+    snapshot.replaceWindowsPaths,
+  ),
+  spec: snapshot.transform(
+    replaceSpecDuration,
+    snapshot.replaceWindowsLineEndings,
+    snapshot.replaceStackTrace,
+    snapshot.replaceWindowsPaths,
+  ),
+  dot: snapshot.transform(
+    replaceSpecDuration,
+    snapshot.replaceWindowsLineEndings,
+    snapshot.replaceStackTrace,
+    snapshot.replaceWindowsPaths,
+  ),
+};
+
+describe('test runner output', { concurrency: true }, async () => {
+  for (const reporter of ['dot', 'junit', 'spec', 'tap']) {
+    await it(reporter, async () => {
+      await snapshot.spawnAndAssert(
+        fixtures.path('test-runner/output/test-runner-force-exit.js'),
+        transformers[reporter] ?? defaultTransform,
+        { flags: ['--test-force-exit', `--test-reporter=${reporter}`] },
```

Comment:
The tests are still not using `--test-reporter-destination=file`. I built this branch locally and the following still seems to not work:

```
./node --test --test-force-exit --test-reporter=tap --test-reporter-destination=test.out ./test/fixtures/test-runner/output/test-runner-force-exit.js
```

Reply:
The output file doesn't really matter.

If you run `node --test --test-force-exit --test-reporter=junit ./test/fixtures/test-runner/output/test-runner-force-exit.js`, it'll have the same output as the output file with `node --test --test-force-exit --test-reporter-destination=test.out --test-reporter=junit ./test/fixtures/test-runner/output/test-runner-force-exit.js`

Reply:
> What do you mean 'not work'?

I mean the output written to the file is still truncated.

Reply:
Ahh I see it now:

The following isn't appended:
```
# tests 2
# suites 1
# pass 1
# fail 1
# cancelled 0
# skipped 0
# todo 0
# duration_ms 49.866251
```

Reply:
FWIW, for these file destinations, the test runner creates them, so I think we should be able to properly close and flush them using logic similar to what you already have.

Reply:
I figured out the issue. I'm gonna resolve conflicts and push shortly.